src/lib.rs

```diff
@@ -107,28 +107,39 @@ assert_unique_used_features! {"llvm4-0", "llvm5-0", "llvm6-0", "llvm7-0", "llvm8
 
 /// Defines the address space in which a global will be inserted.
 ///
+/// The default address space is zero. The first six address spaces are available as associated
+/// constants. To convert any higher number, use the `TryFrom` instance. An Address space is a
+/// 24-bit number.
+///
 /// # Remarks
 /// See also: https://llvm.org/doxygen/NVPTXBaseInfo_8h_source.html
 #[derive(Debug, PartialEq, Eq, Copy, Clone)]
-pub enum AddressSpace {
-    Generic = 0,
-    Global = 1,
-    Shared = 3,
-    Const = 4,
-    Local = 5,
+pub struct AddressSpace(u32);
+
+impl AddressSpace {
+    pub const ZERO: Self = Self(0);
+    pub const ONE: Self = Self(1);
+    pub const TWO: Self = Self(2);
+    pub const THREE: Self = Self(3);
+    pub const FOUR: Self = Self(4);
+    pub const FIVE: Self = Self(5);
```

Comment:
These constants are probably redundant. Do you need them?

Reply:
Another alternative is not using magic numbers and having an enum in your own front-end with proper descriptive names for each address space.
`inkwell::AddressSpace::try_from(crate::AddressSpace::Stack)?` is not ugly.